client/me/account/toggle-sites-as-landing-page.tsx

```diff
@@ -0,0 +1,53 @@
+import { ToggleControl } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { useDispatch, useSelector } from 'react-redux';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { successNotice } from 'calypso/state/notices/actions';
+import { savePreference } from 'calypso/state/preferences/actions';
+import { getPreference, isSavingPreference } from 'calypso/state/preferences/selectors';
+
+type ToggleSitesLandingPageProps = {
+	onToggle?: ( checked: boolean ) => void;
+	className?: string;
+};
+
+function ToggleSitesAsLandingPage( { onToggle, className }: ToggleSitesLandingPageProps ) {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+	const useSitesAsLandingPage = useSelector(
+		( state ) => getPreference( state, 'sites-landing-page' )?.useSitesAsLandingPage
+	);
+	const isSaving = useSelector( ( state ) => isSavingPreference( state ) );
+
+	async function handleToggle( isChecked: boolean ) {
+		const preference = { useSitesAsLandingPage: isChecked, updatedAt: Date.now() };
+		dispatch(
+			recordTracksEvent( 'calypso_sites_as_landing_page_toggled', {
+				sites_as_landing_page: isChecked,
+			} )
+		);
+
+		await dispatch( savePreference( 'sites-landing-page', preference ) );
+
+		dispatch(
+			successNotice( translate( 'Settings saved successfully!' ), {
```

Comment:
Do we need this notice? I think the fact that the toggle is now active is enough, visually, to indicate that the preference was saved 🤔

Reply:
Doesn't seem super necessary, but it's consistent with the other interface settings at least. One benefit might be to avoid confusion for users who are looking for a save button. Some settings have a save button and some don't, so maybe it's good settings that update immediately to have this notice to prevent users looking for something that doesn't exist.

Reply:
I don't know if that's the real reason, just my posthoc rationalisation 😅 I'm not that attached to the notice.

Reply:
I'm of the save view here as @p-jackson, since there's no save button in this UI I assumed the notification is to indicate that the action was saved. The other components here shows the notice so I keep it the same.

Reply:
Fair enough, let's keep it!